blog/src/__tests__/BlogDetails.test.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { screen } from '@testing-library/react';
+import { MemoryRouter } from 'react-router-dom';
+import { renderWithProviders } from '../utils/wrappertesting';
+import BlogDetails from '../components/BlogDetails/BlogDetails';
+import { greater } from '../utils/helperfunctions';
+
+describe('Unit tests for BlogDetails Components', () => {
+  it('Snapshot test for BlogDetails component', () => {
+    const { asFragment } = renderWithProviders(<MemoryRouter><BlogDetails /></MemoryRouter>);
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('Dom test for rendering BlogDetails', () => {
+    renderWithProviders(<MemoryRouter><BlogDetails /></MemoryRouter>);
+    const detailseElement = screen.getByTestId('details');
+    expect(detailseElement).toBeInTheDocument();
+  });
+
+  it('Image renders in BlogDetails component', () => {
+    renderWithProviders(<MemoryRouter><BlogDetails blogDetails={greater[0]} key={greater.id} /></MemoryRouter>);
+    const image = screen.getByTestId('Image');
+    expect(image).toBeInTheDocument();
+  });
+
+  it('Image have ALT', () => {
+    renderWithProviders(<MemoryRouter><BlogDetails blogDetails={greater[0]} key={greater.id} /></MemoryRouter>);
+    const img = screen.getByTestId('Image');
+    expect(img).toHaveAttribute('alt', 'Article');
+  });
```

Comment:
It would be awesome to add an Alt attribute for respective blogs. This helps in SEO. For Example, Alt for a blog named Future of Crypto Currency can be `Future of Crypto Currency Article Header`.
Here you can check if the proper alt attribute for the blog is sent in the blogDetails prop.

This would be the better structure for the Blog object:
```js
{
    id: 2,
    title: '10 Tips for Budget-Friendly Travel',
    author: 'Elena Rodriguez',
    authorImage: {src: images.author1, alt: "10 Tips for Budget-Friendly Travel Author Image"},
    date: '15 August',
    image:{src: images.blogImg1, alt: "10 Tips for Budget-Friendly Travel Blog Header Image"},
    verified: true,
    slug: 'post1',
    content: "test Traveling on a  ",
  }
```